### PR TITLE
[Format]ORC Support ZStandard compression using zstd-jni

### PR DESCRIPTION
**Pull request by zyl891229**
### Purpose
Support ZStandard compression using zstd-jni.

ORC Original PR: https://github.com/apache/orc/pull/988
Support the use of [zstd-jni](https://github.com/luben/zstd-jni) library as the implementation of ORC zstd, 
with better performance than [aircompressor](https://github.com/airlift/aircompressor). 

But ORC support it from 2.0.0 ,but it also drop java 8 at the same version.

### Tests
org.apache.paimon.format.orc.writer.OrcZstdTest

### CODE REVIEW

**Comment by zyl891229** [comment]
Hi @JingsongLi  
According to orc's official website releases list 
1.9.2 is the newest stable version for java 8
image

**Review by JingsongLi — APPROVED** [review]
+1

File: paimon-format/src/main/java/org/apache/orc/OrcConf.java [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/** Define the configuration properties that Orc understands. */
+public enum OrcConf {
```
Comment (Zouxxyy) [inline paimon-format/src/main/java/org/apache/orc/OrcConf.java:29]: I encountered the class conflict in my local test, there may be problems with class with the same name.

```
Caused by: java.lang.NoSuchFieldError: COMPRESSION_ZSTD_LEVEL
	at org.apache.paimon.format.OrcOptions.<clinit>(OrcOptions.java:42)
	at org.apache.paimon.format.orc.OrcWriterFactory.create(OrcWriterFactory.java:132)
	at org.apache.paimon.io.SingleFileWriter.<init>(SingleFileWriter.java:68)
	at org.apache.paimon.io.StatsCollectingSingleFileWriter.<init>(StatsCollectingSingleFileWriter.java:57)
	at org.apache.paimon.io.RowDataFileWriter.<init>(RowDataFileWriter.java:57)
	at org.apache.paimon.io.RowDataRollingFileWriter.lambda$new$0(RowDataRollingFileWriter.java:53)
	at org.apache.paimon.io.RollingFileWriter.openCurrentWriter(RollingFileWriter.java:99)
	at org.apache.paimon.io.RollingFileWriter.write(RollingFileWriter.java:78)
	at org.apache.paimon.append.AppendOnlyWriter$DirectSinkWriter.write(AppendOnlyWriter.java:347)
	at org.apache.paimon.append.AppendOnlyWriter.write(AppendOnlyWriter.java:144)
	at org.apache.paimon.append.AppendOnlyWriter.write(AppendOnlyWriter.java:60)
	at org.apache.paimon.operation.AbstractFileStoreWrite.write(AbstractFileStoreWrite.java:126)
	at org.apache.paimon.table.sink.TableWriteImpl.writeAndReturn(TableWriteImpl.java:145)
	at org.apache.paimon.table.sink.TableWriteImpl.write(TableWriteImpl.java:126)
```

Comment (JingsongLi) [reply paimon-format/src/main/java/org/apache/orc/OrcConf.java:29]: https://github.com/apache/paimon/commit/0be4a21161fd1c6bc957907c9418e2b56417a111 fixed this.

**Pull request merged**